content/techniques/validation.md

```diff
@@ -254,6 +254,31 @@ findOne(
 
 > info **Hint** The `ParseIntPipe` and `ParseBoolPipe` are exported from the `@nestjs/common` package.
 
+If you are using a class to capture more than one parameter or query you can use the `@Type` decorator from `class-transformer` to transform the inpu
+
+```typescript
+class QueryDTO {
+  @IsBoolean()
+  @Type(() => Boolean)
+  public sort: boolean;
+}
+
+class ParamDTO {
+  @IsInt()
+  @Type(() => Number)
+  public id: number;
+}
+
+
+@Get(':id')
+findOne(
+  @Param() params: ParamDTO,
+  @Query() query: QueryDTO,
+) {
+  return 'This action returns a user';
+}
+```
```

Comment:
To make this work properly, you would have to turn on the `transform` configuration in the `ValidationPipe`, which is disabled by default.

Reply:
This will still allow the `IsBoolean` and `IsInt` validations to run inside the pipe though. Regardless of if `transform: true` is set or not, cause we do call `plainToClass` and these `@Transform()` decorators will be handled then

Reply:
What I'm trying to say is that in case `transform` is disabled and there are no other configurations passed into the `ValidationPipe` (`class-validator` specific), the pipe will eventually return the original object (without any transformations applied, see line 133 in `validation.pipe.ts`) so saying:

> If you are using a class to capture more than one parameter or query you can use the `@Type` decorator from `class-transformer` to transform the input

Is partially incorrect, since with the default options the controller won't receive the transformer input, but rather the original one.

Reply:
Ah, gotcha, definitely correct in that case. Maybe we can add the "if `transform: true` is set for the pipe options" (or something similar )to the sentance